src/org/hammerhead226/sharkmacro/motionprofiles/ProfileHandler.java

```diff
@@ -147,71 +187,157 @@ public void manage() {
 			}
 			break;
 		case STARTED:
-			if (status.btmBufferCnt > Constants.MINIMUM_POINTS_IN_TALON) {
+			if (leftStatus.btmBufferCnt > Constants.MINIMUM_POINTS_IN_TALON
+					&& rightStatus.btmBufferCnt > Constants.MINIMUM_POINTS_IN_TALON) {
 				setMode(SetValueMotionProfile.Enable);
 				executionState = ExecutionState.EXECUTING;
 			}
 			break;
 		case EXECUTING:
-			if (status.activePointValid && status.isLast) {
+			if ((leftStatus.activePointValid && leftStatus.isLast)
+					&& (rightStatus.activePointValid && rightStatus.isLast)) {
 				onFinish();
 			}
 			break;
 		}
 	}
 
 	/**
-	 * Sets the state of {@link #talon}.
+	 * Sets the state of the talons.
 	 * 
 	 * @param t
 	 *            the motion profile mode to set the Talon to
 	 */
 	private void setMode(SetValueMotionProfile mode) {
 		this.currentMode = mode;
-		talon.set(ControlMode.MotionProfile, mode.value);
+		leftTalon.set(ControlMode.MotionProfile, mode.value);
+		rightTalon.set(ControlMode.MotionProfile, mode.value);
 	}
 
-	
 	/**
 	 * Fill the Talon's top-level buffer with a given motion profile.
 	 * 
 	 * @param gainsProfile
 	 *            the PID gains profile to use to execute the motion profile
 	 */
-	private void fillTalonWithMotionProfile(int gainsProfile) {
+	private void fillTalonsWithMotionProfile() {
 
+		// maybe need two point objects?
 		TrajectoryPoint point = new TrajectoryPoint();
 
-		if (profileIndex == 0) {
-			talon.clearMotionProfileTrajectories();
-			talon.configMotionProfileTrajectoryPeriod(TrajectoryDuration.Trajectory_Duration_0ms.value, 0);
-			talon.clearMotionProfileHasUnderrun(0);
+		if (leftProfileIndex == 0) {
+			leftTalon.clearMotionProfileTrajectories();
+			leftTalon.configMotionProfileTrajectoryPeriod(TrajectoryDuration.Trajectory_Duration_0ms.value, 0);
+			leftTalon.clearMotionProfileHasUnderrun(0);
+		}
+
+		if (rightProfileIndex == 0) {
+			rightTalon.clearMotionProfileTrajectories();
+			rightTalon.configMotionProfileTrajectoryPeriod(TrajectoryDuration.Trajectory_Duration_0ms.value, 0);
+			rightTalon.clearMotionProfileHasUnderrun(0);
+		}
+
+		while (leftStatus.topBufferCnt < Constants.TALON_TOP_BUFFER_MAX_COUNT
+				&& leftProfileIndex < leftProfile.length) {
+
+			point.position = leftProfile[leftProfileIndex][0];
+			point.velocity = leftProfile[leftProfileIndex][1];
+			point.headingDeg = 0;
+			point.timeDur = toTrajectoryDuration((int) leftProfile[leftProfileIndex][2]);
+			point.profileSlotSelect0 = leftGainsProfile;
+			point.profileSlotSelect1 = 0;
+			point.zeroPos = false;
+			if (leftProfileIndex == 0) {
+				point.zeroPos = true;
```

Comment:
(Of course it's named this and not `isFirstPoint` for consistency...)

Reply:
👍 👍